app/components/button_component.rb

```diff
@@ -18,12 +24,17 @@ def initialize(
     @confirm = confirm
     @disabled = disabled
     @turbo_stream = turbo_stream
+    @scheme = scheme
 
     super(data: data, **kwargs)
   end
 
   private
 
+  def classes
+    (base_classes + send(SCHEME_MAPPINGS[@scheme])).join(" ")
+  end
```

Comment:
Does this mean we can't overload `@classes` anymore to fine-tune margin or padding for the particular usage?

Reply:
Not if you're using a `scheme`, no. My thinking is that eventually we'll add e.g. `size` variants, which will set different paddings, font size, etc, and be able to combine a scheme with a size.

Reply:
More generally, I'd prefer we start with relatively rigid components with a limited number of variants. It feels like we should not need to be fine-tuning things like a button's padding on a one-off basis.